src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1289,4 +1290,1373 @@ void EIRPlantLoopHeatPump::oneTimeInit(EnergyPlusData &state)
+                                                                                     NodeInputManager::CompFluidStream::Secondary,
+                                                                                     DataLoopNode::ObjectIsNotParent);
+                thisPLHP.sourceSideNodes.outlet = NodeInputManager::GetOnlySingleNode(state,
+                                                                                      sourceSideOutletNodeName,
+                                                                                      nodeErrorsFound,
+                                                                                      objType,
+                                                                                      thisPLHP.name,
+                                                                                      condenserNodeType,
+                                                                                      condenserNodeConnectionType_Outlet,
+                                                                                      NodeInputManager::CompFluidStream::Secondary,
+                                                                                      DataLoopNode::ObjectIsNotParent);
+                if (nodeErrorsFound) errorsFound = true;
+                BranchNodeConnections::TestCompSet(
+                    state, cCurrentModuleObject, thisPLHP.name, loadSideInletNodeName, loadSideOutletNodeName, classToInput.nodesType);
+
+                // if (thisPLHP.waterSource) {
+                //    BranchNodeConnections::TestCompSet(
+                //        state, cCurrentModuleObject, thisPLHP.name, sourceSideInletNodeName, sourceSideOutletNodeName, "Condenser Water Nodes");
+                //}
+
+                // store the worker functions that generalized the heating/cooling sides
+                thisPLHP.calcLoadOutletTemp = classToInput.calcLoadOutletTemp;
+                thisPLHP.calcQsource = classToInput.calcQsource;
+                thisPLHP.calcSourceOutletTemp = classToInput.calcSourceOutletTemp;
+
+                if (!errorsFound) {
+                    state.dataEIRFuelFiredHeatPump->heatPumps.push_back(thisPLHP);
+                }
+            }
+        }
+    }
+    if (errorsFound) {
+        // currently there are no straightforward unit tests possible to get here
+        // all curves are required and inputs are validated by the input processor
+        // obviously this will stay here but I don't feel like counting it against coverage
+        ShowFatalError(state, "Previous EIR PLFFHP errors cause program termination."); // LCOV_EXCL_LINE
+    }
+}
+
+void EIRFuelFiredHeatPump::oneTimeInit(EnergyPlusData &state)
+{
+    // This function does all the one-time initialization
+    std::string static const routineName = std::string("EIRFuelFiredHeatPump :") + __FUNCTION__;
```

Comment:
Just do a string with the name you want. This is too cute. `__FUNCTION__` is for putting into other macros that you want to call from different functions.

Reply:
Done